app/models/tariff_update.rb

```diff
@@ -4,62 +4,48 @@ class TariffUpdate
 
   collection_path '/admin/updates'
 
-  attributes :update_type, :state, :issue_date, :created_at, :updated_at, :applied_at, :filesize,
-             :exception_backtrace, :exception_class, :exception_queries,
-             :file_presigned_url, :log_presigned_urls, :presence_errors
+  attributes :update_type,
+             :state,
+             :issue_date,
+             :created_at,
+             :updated_at,
+             :applied_at,
+             :filesize,
+             :exception_backtrace,
+             :exception_class,
+             :exception_queries,
+             :file_presigned_url,
+             :presence_errors
 
   STATES = {
     'A' => 'Applied',
-    'M' => 'Missing',
     'P' => 'Pending',
     'F' => 'Failed',
   }.freeze
 
+  ROLLBACK_APPLICABLE_STATES = %w[A P].freeze
+
   def state
     STATES[super]
   end
 
-  def update_type
-    case attributes[:update_type]
-    when /Taric/ then 'TARIC'
-    when /Chief/ then 'CHIEF'
-    when /Cds/   then 'CDS'
-    end
-  end
-
-  def log_presigned_urls
-    attributes[:log_presigned_urls] || {}
-  end
-
-  def missing?
-    attributes[:state] == 'M'
+  def inserts
+    JSON.parse(attributes[:inserts].presence || '{}')
   end
 
-  def filename
-    super unless missing?
+  def rollback?
+    attributes[:state].in?(ROLLBACK_APPLICABLE_STATES)
   end
 
-  def file_date
-    if update_type == 'CDS'
-      Date.parse(issue_date)
-    else
-      filename.try :slice, 0, 10
-    end
-  end
-
-  def created_at
-    Time.zone.parse(super)
+  def issue_date
+    Date.parse(attributes[:issue_date])
   end
 
   def applied_at
     Time.zone.parse(super) if super.present?
   end
 
-  def to_s
-    "Applied #{update_type} at #{updated_at} (#{filename})"
-  end
-
   def id
-    created_at.to_s.parameterize
+    created_at.parameterize
```

Comment:
this should continue to include `to_s` - if for any reason `created_at` isn't assigned this will blow up, also if created_at is a Date or Time it'll error

Reply:
Not sure about this. It sounds defensive for something that is under our control. If this is nil I'd like it to explode and let me know